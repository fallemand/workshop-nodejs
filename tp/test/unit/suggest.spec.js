const React = require('react');
const renderer = require('react-test-renderer');
const Suggest = require('../../views/components/Suggest');

describe('Suggest (Snapshot)', () => {
    // Metodo de testeo, Ãºtil para componentes
    const component = renderer.create(<Suggest query="iph" />);
    const response = {suggested_queries: [
        {q: 'iphone'},
        {q: 'iphone 5'},
        {q: 'iphone 6'}
    ]};

    afterEach(() => {
        const instance = component.getInstance();
        instance.setState(() => {
            results: []
        });
    });

    test('Suggest render without results',()=> {
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    test('Suggest render with results [change state]',()=> {
        const instance = component.getInstance();
        instance.setState({
            suggested_queries: [{ q: 'iphone', q: 'iphone 6'}]
        });
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    test('Suggest render with results [componentWillReceiveProps]',()=> {
        const instance = component.getInstance();
        window.fetch = jest.fn().mockImplementation(
            () => Promise.resolve({json: () => Promise.resolve({
                suggested_queries: [{ q: 'iphone', q: 'iphone 6'}]
            })})
        );
        expect.assertions(1);
        return instance.componentWillReceiveProps({query: 'iph'}).then(() => {
            const json = component.toJSON();
            expect(json).toMatchSnapshot();
        });
    });
});