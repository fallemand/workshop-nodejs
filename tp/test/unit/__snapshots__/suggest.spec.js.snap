// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest (Snapshot) Suggest render with results [change state] 1`] = `<div />`;

exports[`Suggest (Snapshot) Suggest render with results [componentWillReceiveProps] 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 6"
      >
        iphone 6
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest (Snapshot) Suggest render without results 1`] = `<div />`;
